File: rust-core/tests/test_repos/simple_python_project/math_utils.py
class MathUtils:
    """Class for mathematical operations"""
    
    def calculate_sum(self, numbers):
        """Calculate sum of numbers"""
        if not numbers:
            return 0
        
        total = 0
        for num in numbers:
            total = self._add_to_total(total, num)
        return total
    
    def _add_to_total(self, current_total, new_number):
        """Add a new number to the current total"""
        return current_total + new_number
    
    def calculate_average(self, numbers):
        """Calculate average of numbers"""
        if not numbers:
            return 0
        
        total = self.calculate_sum(numbers)
        count = len(numbers)
        return self._divide_numbers(total, count)
    
    def _divide_numbers(self, numerator, denominator):
        """Divide two numbers"""
        if denominator == 0:
            raise ValueError("Cannot divide by zero")
        return numerator / denominator
    
    def factorial(self, n):
        """Calculate factorial of n"""
        if n < 0:
            raise ValueError("Factorial not defined for negative numbers")
        if n <= 1:
            return 1
        return n * self.factorial(n - 1)
    
    def fibonacci(self, n):
        """Calculate nth Fibonacci number"""
        if n <= 0:
            return 0
        if n == 1:
            return 1
        return self.fibonacci(n - 1) + self.fibonacci(n - 2)
    
    def power(self, base, exponent):
        """Calculate base raised to exponent"""
        if exponent == 0:
            return 1
        if exponent < 0:
            return 1 / self.power(base, -exponent)
        
        result = 1
        for _ in range(exponent):
            result = self._multiply_numbers(result, base)
        return result
    
    def _multiply_numbers(self, a, b):
        """Multiply two numbers"""
        return a * b 

File: rust-core/tests/test_repos/simple_js_project/mathUtils.js
class MathUtils {
    calculateSum(numbers) {
        if (!Array.isArray(numbers) || numbers.length === 0) {
            return 0;
        }
        
        let total = 0;
        for (const num of numbers) {
            total = this._addToTotal(total, num);
        }
        return total;
    }
    
    _addToTotal(currentTotal, newNumber) {
        return currentTotal + newNumber;
    }
    
    calculateAverage(numbers) {
        if (!Array.isArray(numbers) || numbers.length === 0) {
            return 0;
        }
        
        const total = this.calculateSum(numbers);
        const count = numbers.length;
        return this._divideNumbers(total, count);
    }
    
    _divideNumbers(numerator, denominator) {
        if (denominator === 0) {
            throw new Error('Cannot divide by zero');
        }
        return numerator / denominator;
    }
    
    factorial(n) {
        if (n < 0) {
            throw new Error('Factorial not defined for negative numbers');
        }
        if (n <= 1) {
            return 1;
        }
        return n * this.factorial(n - 1);
    }
    
    fibonacci(n) {
        if (n <= 0) {
            return 0;
        }
        if (n === 1) {
            return 1;
        }
        return this.fibonacci(n - 1) + this.fibonacci(n - 2);
    }
    
    power(base, exponent) {
        if (exponent === 0) {
            return 1;
        }
        if (exponent < 0) {
            return 1 / this.power(base, -exponent);
        }
        
        let result = 1;
        for (let i = 0; i < exponent; i++) {
            result = this._multiplyNumbers(result, base);
        }
        return result;
    }
    
    _multiplyNumbers(a, b) {
        return a * b;
    }
    
    isPrime(n) {
        if (n < 2) {
            return false;
        }
        if (n === 2) {
            return true;
        }
        if (n % 2 === 0) {
            return false;
        }
        
        for (let i = 3; i <= Math.sqrt(n); i += 2) {
            if (n % i === 0) {
                return false;
            }
        }
        return true;
    }
}

module.exports = MathUtils; 